en/docs/references/extend/federation/write-a-custom-oauth2-federated-authenticator.md

```diff
@@ -0,0 +1,84 @@
+# Write a Custom OAuth2 Federated Authenticator
+
+A custom OAuth2 authenticator can be written to authenticate a user with an external system. The external system can be
+any Identity provider supporting OAuth2 including Facebook, Kakao, Naver etc. You can use the OAuth2GenericAuthenticator
+available in WSO2 Identity Server to create custom OAuth2 authenticators.
+
+## Steps to write a custom OAuth2 authenticator
+
+1. First create a maven project for the custom OAuth2 authenticator. Refer to
+   the [pom.xml](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.oauth2.federated.authenticator/pom.xml)
+   file used for the Custom authenticator.
+2. Refer
+   the [service component class](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.oauth2.federated.authenticator/src/main/java/org/wso2/carbon/identity/sample/oauth2/federated/authenticator/internal/OAuth2CustomAuthenticatorServiceComponent.java)
+   as well since the authenticator is written as an OSGI service to deploy in WSO2 Identity Server and register it as an
+   authenticator
```

Comment:
```suggestion
1. Create a maven project for the custom OAuth2 authenticator. Refer to the following files:

> - the [pom.xml](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.oauth2.federated.authenticator/pom.xml) file used for the Custom authenticator.
> - The [service component class](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.oauth2.federated.authenticator/src/main/java/org/wso2/carbon/identity/sample/oauth2/federated/authenticator/internal/OAuth2CustomAuthenticatorServiceComponent.java) as well since the authenticator is written as an OSGI service to deploy in WSO2 Identity Server and register it as an authenticator
```